cli/impl/src/main/java/io/helidon/build/cli/impl/InitCommand.java

```diff
@@ -93,6 +93,9 @@ protected void invoke(CommandContext context) throws Exception {
             assertSupportedVersion(helidonVersion);
         }
 
+        String cliVersion = getClass().getPackage().getImplementationVersion();
```

Comment:
The current thinking is that the cli will only perform the check using the new XML file, and the schema version check only inside the engine.

Reply:
To do check using new XML file we need two values : CLI version and Helidon archetype version, right? Then using rules from the new XML file we check the compatibility of these two values, right? So what is wrong with this line?

Reply:
`versions.xml` will be used primarily to resolve the latest version, and the filter the list of versions. We can also perform a check in case the version is supplied explicitly.

However the true compatibility check that involves schema versions should be done only inside the engine.